c7n/resources/vpc.py

```diff
@@ -2708,3 +2708,80 @@ class resource_type(query.TypeInfo):
         arn_type = 'traffic-mirror-target'
         universal_taggable = object()
         id_prefix = 'tmt-'
+
+
+@RouteTable.filter_registry.register('cross-az-nat-gateway-route')
+class CrossAZRouteTable(Filter):
+    """Filter route-table which is attached to a subnet that is not in the same availability zone (AZ) of the
+    NAT Gateway subnet AZ. This filter is applicable to those route-tables which have next hop as NatGateways for
+    any route. If a route-table has NAT gateway from us-west-2a and it is associated to subnet from us-west-2b then it
+    will flagged, and NatGatewayInCrossAvailabilityZone key would be set a True.
+
+    :Example:
+    .. code-block:: yaml
+            policies:
+              - name: cross-az-nat-gateway--traffic
```

Comment:
@vvishwakarma-intuit this sample policy can't be deployed because it has a bad policy name. Change `cross-az-nat-gateway--traffic` to `cross-az-nat-gateway-traffic`

Reply:
corrected the policy name

Reply:
corrected the policy name